modules/foreman_host.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2017, Ismael Puerto <ipuertofreire@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_host
+short_description: Manage Foreman Host
+description:
+    - Manage host
+author: "Ismael Puerto (@ismaelpuerto)"
+requirements:
+    - "nailgun >= 0.28.0"
+    - "python >= 2.6"
+options:
+    server_url:
+        description:
+            - URL of Foreman server
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+        required: false
+    hostname:
+        description:
+            - Hostname
+        required: true
+    mac:
+        description:
+            - MAC HOST
+        required: true
+    organization:
+        description:
+            - Organization
+        required: true
+    location:
+        description:
+            - Location
+        required: true
+    architecture
+        description:
+            - Architecture
+        required: true
+    operatingsystem
+        description:
+            - Operating System
+        required: true
+'''
+
+EXAMPLES = '''
+- name: "Create Host"
+  foreman_host:
+    username: "admin"
+    password: "admin"
+    server_url: "https://foreman.example.com"
+    hostname: "hostname"
+    mac: "00:00:00:00:00:00"
+    organization: "Default Organization"
+    location: "Default Location"
+    architecture: "x86_64"
+    operatingsystem: "Fedora 26"
+  delegate_to: localhost
+'''
+
+RETURN = '''# '''
+
+try:
+    from nailgun.entities import (
+            Host,
+    )
+    from ansible.module_utils.ansible_nailgun_cement import (
+            create_server,
+            create_entity,
+            update_entity,
+            handle_no_nailgun,
+            find_organization,
+            find_operatingsystem,
+            find_architecture,
+            find_location,
+            ping_server,
+    )
+    HAS_NAILGUN_PACKAGE = True
+except:
+    HAS_NAILGUN_PACKAGE = False
+
+def create_host(module, hostname, mac, organization, location, architecture, operatingsystem):
```

Comment:
This name is misleading, as a host is not always created, but at times just updated. Also, for managing, deleting and changing the powerstate would be interesting.